#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -1 0 0 1.5707963267948966
  position 0.0005558355844535609 5.697026724861683 1.2613036905340174e-06
}
Background {
  skyColor [
    1 1 1
  ]
}
DirectionalLight {
  ambientIntensity 1
  direction 0.1 -0.5 0.3
}
RectangleArena {
  name "arena"
  floorSize 10 10
  floorTileSize 2 2
  floorAppearance PBRAppearance {
    baseColorMap ImageTexture {
      url [
        "textures/parquetry/chequered_parquetry_base_color.jpg"
      ]
    }
    roughness 1
    roughnessMap ImageTexture {
      url [
        "textures/parquetry/chequered_parquetry_roughness.jpg"
      ]
    }
    metalness 0
    normalMap ImageTexture {
      url [
        "textures/parquetry/chequered_parquetry_normal.jpg"
      ]
    }
    occlusionMap ImageTexture {
      url [
        "textures/parquetry/chequered_parquetry_occlusion.jpg"
      ]
    }
  }
  wallHeight 0.3
  wallAppearance PBRAppearance {
    baseColor 0.666667 0 0
    roughness 1
    metalness 0
  }
}
Robot {
  children [
    Group {
    }
    Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        roughness 0.3
        metalness 0
      }
      geometry IndexedFaceSet {
        coord Coordinate {
          point [
            -0.1 0.059 0.215
            0.1 0.059 0.215
            0.135 0.059 0.185
            0.135 0.059 -0.095
            0.08 0.059 -0.11
            -0.08 0.059 -0.11
            -0.135 0.059 -0.095
            -0.135 0.059 0.185
            -0.1 0.059 0.215
            -0.1 0.234 0.215
            0.1 0.234 0.215
            0.135 0.234 0.185
            0.135 0.234 -0.095
            0.08 0.234 -0.11
            -0.08 0.234 -0.11
            -0.135 0.234 -0.095
            -0.135 0.234 0.185
            -0.1 0.234 0.215
          ]
        }
        coordIndex [
          8, 7, 6, 5, 4, 3, 2, 1, 0, -1, 9, 10, 11, 12, 13, 14, 15, 16, 17, -1, 0, 1, 10, 9, -1, 1, 2, 11, 10, -1, 2, 3, 12, 11, -1, 3, 4, 13, 12, -1, 4, 5, 14, 13, -1, 5, 6, 15, 14, -1, 6, 7, 16, 15, -1, 7, 8, 17, 16, -1
        ]
      }
    }
    Shape {
      appearance PBRAppearance {
        baseColor 0.636415 0.655556 0.655556
        roughness 0.3
        metalness 0
      }
      geometry IndexedFaceSet {
        coord Coordinate {
          point [
            0 0.234 -0.174, -0.056 0.234 -0.166, -0.107 0.234 -0.145, -0.155 0.234 -0.112, -0.19 0.234 -0.064, -0.19 0.234 0.074, -0.16 0.234 0.096, -0.16 0.234 0.151, -0.155 0.234 0.2, -0.107 0.234 0.236, -0.056 0.234 0.256, 0 0.234 0.264, 0.056 0.234 0.256, 0.107 0.234 0.236, 0.155 0.234 0.2, 0.16 0.234 0.151, 0.16 0.234 0.096, 0.19 0.234 0.074, 0.19 0.234 -0.064, 0.155 0.234 -0.112, 0.107 0.234 -0.145, 0.056 0.234 -0.166, 0 0.234 -0.174, 0 0.245 -0.174, -0.056 0.245 -0.166, -0.107 0.245 -0.145, -0.155 0.245 -0.112, -0.19 0.245 -0.064, -0.19 0.245 0.074, -0.16 0.245 0.096, -0.16 0.245 0.151, -0.155 0.245 0.2, -0.107 0.245 0.236, -0.056 0.245 0.256, 0 0.245 0.264, 0.056 0.245 0.256, 0.107 0.245 0.236, 0.155 0.245 0.2, 0.16 0.245 0.151, 0.16 0.245 0.096, 0.19 0.245 0.074, 0.19 0.245 -0.064, 0.155 0.245 -0.112, 0.107 0.245 -0.145, 0.056 0.245 -0.166, 0 0.245 -0.174
          ]
        }
        coordIndex [
          22, 21, 20, 19, 18, 17, 16, 15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0, -1, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, -1, 0, 1, 24, 23, -1, 1, 2, 25, 24, -1, 2, 3, 26, 25, -1, 3, 4, 27, 26, -1, 4, 5, 28, 27, -1, 5, 6, 29, 28, -1, 6, 7, 30, 29, -1, 7, 8, 31, 30, -1, 8, 9, 32, 31, -1, 9, 10, 33, 32, -1, 10, 11, 34, 33, -1, 11, 12, 35, 34, -1, 12, 13, 36, 35, -1, 13, 14, 37, 36, -1, 14, 15, 38, 37, -1, 15, 16, 39, 38, -1, 16, 17, 40, 39, -1, 17, 18, 41, 40, -1, 18, 19, 42, 41, -1, 19, 20, 43, 42, -1, 20, 21, 44, 43, -1, 21, 22, 45, 44, -1
        ]
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0 0.0825 0
      }
      device [
        RotationalMotor {
          name "left wheel motor"
        }
        PositionSensor {
          name "left wheel sensor"
        }
      ]
      endPoint Solid {
        translation -0.1603 0.0825 0
        rotation 1 0 0 5.2408332
        children [
          DEF WHEEL Transform {
            rotation 0 0 1 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0.0037037 0.0037037 0.0037037
                  roughness 0.3
                  metalness 0
                }
                geometry Cylinder {
                  height 0.037
                  radius 0.0825
                }
              }
            ]
          }
        ]
        name "left wheel"
        boundingObject USE WHEEL
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0 0.0825 0
      }
      device [
        RotationalMotor {
          name "right wheel motor"
        }
        PositionSensor {
          name "right wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0.1603 0.0825 0
        rotation 0.9999999999999999 0 0 0.9226862300000003
        children [
          USE WHEEL
        ]
        name "right wheel"
        boundingObject USE WHEEL
      }
    }
    Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        roughness 0.3
        metalness 0
      }
      geometry IndexedFaceSet {
        coord Coordinate {
          point [
            0.136 0.184 -0.046
            0.136 0.184 -0.096
            0.101 0.184 -0.134
            0.054 0.184 -0.159
            0 0.184 -0.168
            -0.054 0.184 -0.159
            -0.101 0.184 -0.134
            -0.136 0.184 -0.096
            -0.136 0.184 -0.046
            0.136 0.184 -0.043
            0.136 0.234 -0.046
            0.136 0.234 -0.096
            0.101 0.234 -0.134
            0.054 0.234 -0.159
            0 0.234 -0.168
            -0.054 0.234 -0.159
            -0.101 0.234 -0.134
            -0.136 0.234 -0.096
            -0.136 0.234 -0.046
            0.136 0.234 -0.043
          ]
        }
        coordIndex [
          9, 8, 7, 6, 5, 4, 3, 2, 1, 0, -1, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, -1, 0, 1, 11, 10, -1, 1, 2, 12, 11, -1, 2, 3, 13, 12, -1, 3, 4, 14, 13, -1, 4, 5, 15, 14, -1, 5, 6, 16, 15, -1, 6, 7, 17, 16, -1, 7, 8, 18, 17, -1, 8, 9, 19, 18, -1, 9, 0, 10, 19, -1
        ]
      }
    }
    Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        roughness 0.3
        metalness 0
      }
      geometry IndexedFaceSet {
        coord Coordinate {
          point [
            -0.136 0.184 0.135
            -0.136 0.184 0.185
            -0.101 0.184 0.223
            -0.054 0.184 0.248
            0 0.184 0.257
            0.054 0.184 0.248
            0.101 0.184 0.223
            0.136 0.184 0.185
            0.136 0.184 0.135
            -0.136 0.184 0.135
            -0.136 0.234 0.135
            -0.136 0.234 0.185
            -0.101 0.234 0.223
            -0.054 0.234 0.248
            0 0.234 0.257
            0.054 0.234 0.248
            0.101 0.234 0.223
            0.136 0.234 0.185
            0.136 0.234 0.135
            -0.136 0.234 0.135
          ]
        }
        coordIndex [
          9, 8, 7, 6, 5, 4, 3, 2, 1, 0, -1, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, -1, 0, 1, 11, 10, -1, 1, 2, 12, 11, -1, 2, 3, 13, 12, -1, 3, 4, 14, 13, -1, 4, 5, 15, 14, -1, 5, 6, 16, 15, -1, 6, 7, 17, 16, -1, 7, 8, 18, 17, -1, 8, 9, 19, 18, -1
        ]
      }
    }
    Transform {
      translation 0 0.0325 0.2147
      rotation 0 0 1 1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.0037037 0.0037037 0.0037037
            roughness 0.3
            metalness 0
          }
          geometry Cylinder {
            height 0.024
            radius 0.0325
          }
        }
      ]
    }
    DistanceSensor {
      translation 0 0.209 -0.167
      rotation 0 1 0 1.57
      children [
        DEF SONAR Transform {
          rotation 0 0 1 1.57
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 1 1 0.0517103
                roughness 0.3
                metalness 0
              }
              geometry Cylinder {
                height 0.002
                radius 0.0175
              }
            }
          ]
        }
      ]
      name "ds_front"
      lookupTable [
        0 1024 0
        0.2 1024 0.1
        0.4 0 0.1
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation -0.001 0.209 0.259
      rotation 0 1 0 -1.5699996938995748
      children [
        USE SONAR
      ]
      name "ds_back"
      lookupTable [
        0 1024 0
        0.2 1024 0.1
        0.4 0 0.1
      ]
      type "infra-red"
    }
  ]
  name "crashbot1"
  description "designed by Activmedia Robotics"
  boundingObject Transform {
    translation 0 0.13 0.0447
    children [
      Cylinder {
        height 0.24
        radius 0.219
      }
    ]
  }
  controller "crash_derby_remote_python"
  customData "3001"
}
